src/librustc/lint/mod.rs

```diff
@@ -713,6 +713,16 @@ pub fn struct_lint_level<'a>(sess: &'a Session,
     return err
 }
 
+pub fn maybe_lint_level_root(tcx: TyCtxt<'_, '_, '_>, id: hir::HirId) -> bool {
```

Comment:
Can you rename this to `is_lint_level_root`?

Reply:
Just because this is true, doesn't mean it is an actual lint root (i.e. shows up in the lint level map), hence the maybe.